lib/ex_machina/ecto.ex

```diff
@@ -0,0 +1,58 @@
+defmodule ExMachina.Ecto do
+  defmacro __using__(opts) do
+    quote do
+      use ExMachina
+
+      @repo Dict.get(unquote(opts), :repo)
+
+      def fields_for(factory_name, attrs \\ %{}) do
+        ExMachina.Ecto.fields_for(__MODULE__, factory_name, attrs)
+      end
+
+      def save_record(record) do
+        ExMachina.Ecto.save_record(__MODULE__, @repo, record)
+      end
+    end
+  end
+
+  @doc """
+  Builds a factory with the passed in factory_name and returns its fields
+
+  This is only for use with Ecto models.
+
+  Will return a map with the fields and virtual fields, but without the Ecto
+  metadata and associations.
+
+  ## Example
+
+      def factory(:user) do
+        %MyApp.User{name: "John Doe", admin: false}
+      end
+
+      # Returns %{name: "John Doe", admin: true}
+      fields_for(:user, admin: true)
+  """
+  def fields_for(module, factory_name, attrs \\ %{}) do
+    module.build(factory_name, attrs)
+    |> drop_ecto_fields
+  end
+
+  defp drop_ecto_fields(record = %{__struct__: struct, __meta__: %{__struct__: Ecto.Schema.Metadata}}) do
+    record
+    |> Map.from_struct
+    |> Map.delete(:__meta__)
+    |> Map.drop(struct.__schema__(:associations))
+  end
+  defp drop_ecto_fields(record) do
+    raise ArgumentError, "#{inspect record} is not an Ecto model. Use `build` instead."
+  end
```

Comment:
I'm wondering if `assoc` should be only in Ecto and should return the id automatically. It's only useful for Ecto, because you can just use `build(:association)` for other types of factories. What do you think?